linked_list.py

```diff
@@ -33,48 +23,50 @@ def search(self, val):
                 val: String value associated with the val property of Node
         """
         current = self.head
-        exists = False
-        while current and exists is False:
-            if current.get_info() == val:
-                exists = True
+        while current:
+            if current.val == val:
+                break
             else:
-                current = current.next_node()
+                current = current.pointer
         if not current:
-            raise ValueError("That Node is not in this list")
+            return None
         return current
```

Comment:
is there a "falsy" possible value for current that is not `None`?  If not, then why test.  Just return `current`.  Either it's the node you were seeking, or it's `None`.